packages/console/src/components/Tooltip/index.tsx

```diff
@@ -80,15 +97,36 @@ const Tooltip = <T extends Element>({
     }
 
     return () => tooltipDom?.remove();
-  }, [positionCaculated, tooltipDom]);
+  }, [showUp, tooltipDom]);
+
+  useLayoutEffect(() => {
+    mutate();
+  }, [content, mutate]);
 
-  if (!tooltipDom || !position) {
+  if (!tooltipDom) {
     return null;
   }
 
+  const isArrowDown = positionState.verticalAlign === 'top';
+
   return createPortal(
-    <div className={classNames(styles.container, className)} style={{ ...position }}>
-      {content}
+    <div
+      ref={tooltipRef}
+      className={classNames(
+        styles.tooltip,
+        isSuccessful ? styles.tooltipSuccess : styles.tooltipDefault,
+        className
+      )}
+      style={{ ...position }}
+    >
+      <div
+        className={classNames(
+          styles.arrow,
+          isArrowDown ? styles.arrowDown : styles.arrowUp,
+          isSuccessful ? (isArrowDown ? styles.arrowDownSuccess : styles.arrowUpSuccess) : undefined
```

Comment:
This looks redundant, which is another reason I think we should remove `isSuccessful`

Reply:
Referer to the replay above.

Reply:
Even if we decide to keep `isSuccessful`, we still don't have to do this. We can apply the `successful` class to the container div, and use the `cascade` ability to affect inner elements. E.g.
```css
.tooltip {
  // some styles

  .arrow {
    // styles for arrow head
  }

  &.successful {
    // styles for successful state

    .arrow {
      // styles for arrow head in successful state
    }
  }
}

```

Reply:
updated 👍